addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/protocol/xml/InputWithDabControlXML.java

```diff
@@ -0,0 +1,236 @@
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+
+import java.io.IOException;
+import java.lang.ref.WeakReference;
+
+/**
+ * This class implements the Yamaha Receiver protocol related to DAB tuners which allows to control band and preset.
+ * This control is specific to dual band tuners only.
+ *
+ * Note that yamaha maintains separate presets for each band.
+ *
+ * The XML nodes <DAB><Play_Control><Band>FM</Band></Play_Control></DAB> are used.
+ *
+ * No state will be saved in here, but in {@link DabBandState}, {@link PresetInfoState} and {@link PlayInfoState} instead.
+ *
+ * @author Tomasz Maruszak - [yamaha] Tuner band selection and preset feature for dual band models (RX-S601D)
+ */
+public class InputWithDabControlXML implements InputWithDabBandControl, InputWithPresetControl {
+
+    private Logger logger = LoggerFactory.getLogger(InputWithDabControlXML.class);
+
+    private static final String BAND_FM = "FM";
+    private static final String BAND_DAB = "DAB";
+
+    protected final WeakReference<AbstractConnection> comReference;
+    protected final String inputID;
+
+    private final DabBandStateListener observerForBand;
+    private final PresetInfoStateListener observerForPreset;
+    private final PlayInfoStateListener observerForPlayInfo;
+
+    /**
+     * Need to remember last band state to drive the preset
+     */
+    private DabBandState bandState = null;
+
+    /**
+     * Create a InputWithPlayControl object for altering menu positions and requesting current menu information as well
+     * as controlling the playback and choosing a preset item.
+     * @param inputID The input ID - TUNER is going to be used here.
+     * @param com The Yamaha communication object to send http requests.
+     */
+    public InputWithDabControlXML(String inputID, AbstractConnection com,
+                                  DabBandStateListener observerForBand,
+                                  PresetInfoStateListener observerForPreset,
+                                  PlayInfoStateListener observerForPlayInfo) {
+        this.inputID = inputID;
+        this.comReference = new WeakReference<>(com);
+        this.observerForBand = observerForBand;
+        this.observerForPreset = observerForPreset;
+        this.observerForPlayInfo = observerForPlayInfo;
+
+        if (observerForBand == null && observerForPreset == null && observerForPlayInfo == null) {
+            throw new IllegalArgumentException("At least one observer has to be provided");
+        }
+    }
+
+    /**
+     * Wraps the XML message with the <DAB> tag:
+     * <DAB>message</DAB>.
+     *
+     * @param message XML message
+     * @return
+     */
+    protected String wrInput(String message) {
+        return "<DAB>" + message + "</DAB>";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void update() throws IOException, ReceivedMessageParseException {
+        AbstractConnection com = comReference.get();
```

Comment:
I can not see parallels to the current usage. On stackoverflow they have a CAR, not strongly referenced by anything else, with a normal reference to an ENGINE. If there is only a weak reference to the CAR, of course the CAR can be garbage collected.

The situation here is: We have a HANDLER (strongly referenced by the framework), with a COM instance, strongly referenced by the HANDLER. And we have CONTROLLERS, each which a weak reference to COM. COM will never be garbage collected as long as HANDLER exists.

This is the desired behavior and it protects us from this:
HANDLER disposes(). COM goes away. But:
We didn't clean up well enough and there is a dangling CONTROLLER left. The CONTROLLER tries to access COM, but COM got garbage collected and fails. Perfect. That's what we want.